### ISSUE TITLE
program always rebuilt with -main

### ISSUE THREAD

**Issue by CyberShadow** [dragon-lang/rund#14]
```console
$ echo 'pragma(msg, "Building");' > test.d
$ rund -main test.d
Building
$ rund -main test.d
Building
```

**Comment by CyberShadow** [comment]
```patch
diff --git a/rund.d b/rund.d
index 77b1099..ae4e026 100755
--- a/rund.d
+++ b/rund.d
@@ -22,7 +22,7 @@ import core.stdc.stdlib : exit;
 
 import std.typecons : Flag, Yes, No;
 import std.exception : collectException, enforce;
-import std.algorithm : canFind, map, skipOver, splitter;
+import std.algorithm : canFind, map, filter, skipOver, splitter;
 import std.range : chain, only;
 import std.array : array, appender;
 static import std.string;
@@ -587,7 +587,7 @@ Flag!"compile" determineCompile(bool force, Output output, string jsonFilename,
     // TODO: what happens if json modify time is different from outputFileTime and/or build witness time?
     auto deps = readJsonFile(jsonFilename, objDir);
 
-    auto updated = deps.byKey.anyNewerThan(lastBuildTime);
+    auto updated = deps.byKey.filter!(dep => dep != "__main.d").anyNewerThan(lastBuildTime);
     if (updated)
     {
         yapf("COMPILE(YES) updated file '%s'", updated.formatQuotedIfSpaces);
```

?

**Comment by marler8997** [comment]
Thanks for the fix!

**Comment by marler8997** [comment]
Fixed with #15
